libs/small-steps/src/Data/MemoBytes.hs

```diff
@@ -69,7 +69,7 @@ instance (Typeable t, FromCBOR (Annotator t)) => FromCBOR (Annotator (MemoBytes
     (Annotator getT, Annotator getBytes) <- withSlice fromCBOR
     pure (Annotator (\fullbytes -> Memo (getT fullbytes) (toShort (toStrict (getBytes fullbytes)))))
 
-instance Eq t => Eq (MemoBytes t) where (Memo x _) == (Memo y _) = x == y
+instance Eq (MemoBytes t) where (Memo _ x) == (Memo _ y) = x == y
```

Comment:
Not comparing both constituents of `Memo` implicitly supposes that if two binary representations are equal, then their Haskell representation must be equal a well; is there anything truly enforcing that? While it sounds plausible in my head, is it really the case :thinking: ?

Reply:
This was what I was justifying above:
```
serialise x == serialise y
deserialise (serialise x) == deserialise (serialise y)
x == y
```

The property that `deserialise` is left-inverse to `serialise` _is_ a required property for any serialisable type (as opposed to the other way around!). And we have tests verifying this property for serialised types.